.github/workflows/validation.yml

```diff
@@ -311,6 +305,66 @@ jobs:
             packages/java/gradle-plugin/build/test-results/*
             packages/java/gradle-plugin/build/reports/*
 
+  test-react-18:
```

Comment:
The manual run to verify this change: https://github.com/vaadin/hilla/actions/runs/13588075190